app/controllers/api/v1/users_controller.rb

```diff
@@ -11,6 +11,18 @@ def index
         render json: users, status: :ok
       end
 
+      def destroy_self
+        authorize current_user, :destroy_self?
```

Comment:
Please, check if this works

```suggestion
        authorize
```